diff -Naur mupdf-1.6-source/Makerules mupdf/Makerules
--- mupdf-1.6-source/Makerules	2014-09-30 13:25:12.000000000 +0200
+++ mupdf/Makerules	2015-08-21 10:15:04.000000000 +0200
@@ -33,15 +33,15 @@
 
 HAVE_X11 ?= yes
 
-SYS_OPENSSL_CFLAGS = -DHAVE_OPENSSL
-SYS_OPENSSL_LIBS = -lcrypto
+#SYS_OPENSSL_CFLAGS = -DHAVE_OPENSSL
+#SYS_OPENSSL_LIBS = -lcrypto
 
 SYS_CURL_DEPS = -lpthread
 
-SYS_X11_CFLAGS = -I/usr/X11R6/include
-SYS_X11_LIBS = -L/usr/X11R6/lib -lX11 -lXext
+SYS_X11_CFLAGS = -I/usr/X11/include
+SYS_X11_LIBS = -L/usr/X11/lib -lX11 -lXext
 
-SYS_FREETYPE_CFLAGS = -I/usr/X11R6/include/freetype2
+SYS_FREETYPE_CFLAGS = -I/usr/X11/include/freetype2
 SYS_FREETYPE_LIBS = -lfreetype
 SYS_OPENJPEG_LIBS = -lopenjpeg
 SYS_JBIG2DEC_LIBS = -ljbig2dec
@@ -58,10 +58,10 @@
 
 HAVE_X11 ?= yes
 
-ifeq "$(shell pkg-config --exists libcrypto && echo yes)" "yes"
-SYS_OPENSSL_CFLAGS = -DHAVE_OPENSSL $(shell pkg-config --cflags libcrypto)
-SYS_OPENSSL_LIBS = $(shell pkg-config --libs libcrypto)
-endif
+#ifeq "$(shell pkg-config --exists libcrypto && echo yes)" "yes"
+#SYS_OPENSSL_CFLAGS = -DHAVE_OPENSSL $(shell pkg-config --cflags libcrypto)
+#SYS_OPENSSL_LIBS = $(shell pkg-config --libs libcrypto)
+#endif
 
 ifeq "$(shell pkg-config --exists libcurl && echo yes)" "yes"
 HAVE_CURL = yes
diff -Naur mupdf-1.6-source/Makethird mupdf/Makethird
--- mupdf-1.6-source/Makethird	2014-09-30 13:25:12.000000000 +0200
+++ mupdf/Makethird	2015-08-21 10:15:04.000000000 +0200
@@ -203,6 +203,7 @@
 	jidctfst.c \
 	jidctint.c \
 	jmemmgr.c \
+	jmemnobs.c \
 	jquant1.c \
 	jquant2.c \
 	jutils.c \
@@ -213,7 +214,7 @@
 $(JPEG_OUT)/%.o: $(JPEG_DIR)/%.c | $(JPEG_OUT)
 	$(CC_CMD) -Dmain=xxxmain
 
-JPEG_CFLAGS := -Iscripts/jpeg -I$(JPEG_DIR)
+JPEG_CFLAGS := -Iscripts/jpeg -I$(JPEG_DIR) -DSHARE_JPEG
 else
 JPEG_CFLAGS := $(SYS_JPEG_CFLAGS) -DSHARE_JPEG
 JPEG_LIBS := $(SYS_JPEG_LIBS)
