diff -Naur libvpx-v1.3.0/configure libvpx/configure
--- libvpx-v1.3.0/configure	2014-01-10 21:12:42.000000000 +0100
+++ libvpx/configure	2015-09-17 11:59:06.000000000 +0200
@@ -618,7 +618,7 @@
               # work around them until they are fixed
               check_add_cflags -fno-strict-aliasing
           ;;
-          *) check_add_cflags -Wunused-but-set-variable ;;
+          *) check_add_cflags -Wunused-const-variable ;;
         esac
         enabled extra_warnings || check_add_cflags -Wno-unused-function
     fi
diff -Naur libvpx-v1.3.0/nestegg/halloc/src/align.h libvpx/nestegg/halloc/src/align.h
--- libvpx-v1.3.0/nestegg/halloc/src/align.h	2014-01-10 21:12:40.000000000 +0100
+++ libvpx/nestegg/halloc/src/align.h	2015-09-17 11:55:32.000000000 +0200
@@ -18,7 +18,7 @@
 /*
  *	a type with the most strict alignment requirements
  */
-union max_align
+union _max_align
 {
 	char   c;
 	short  s;
@@ -30,7 +30,7 @@
 	void (*q)(void);
 };
 
-typedef union max_align max_align_t;
+typedef union _max_align _max_align_t;
 
 #endif
 
diff -Naur libvpx-v1.3.0/nestegg/halloc/src/halloc.c libvpx/nestegg/halloc/src/halloc.c
--- libvpx-v1.3.0/nestegg/halloc/src/halloc.c	2014-01-10 21:12:40.000000000 +0100
+++ libvpx/nestegg/halloc/src/halloc.c	2015-09-17 11:57:48.000000000 +0200
@@ -30,7 +30,7 @@
 #endif
 	hlist_item_t  siblings; /* 2 pointers */
 	hlist_head_t  children; /* 1 pointer  */
-	max_align_t   data[1];  /* not allocated, see below */
+	_max_align_t  data[1];  /* not allocated, see below */
 	
 } hblock_t;
 
